ScarletEditor/src/Panels/SceneHierarchy/SceneHierarchy.cpp

```diff
@@ -90,16 +95,79 @@ namespace ScarletEngine
 		}
 	}
 
-	void SceneHierarchyPanel::OnWorldSelectionChanged(const EntityPtr& LastSelectedEntity)
+	void SceneHierarchyPanel::OnWorldSelectionChanged()
```

Comment:
I find it kind of weird that you're looping over all the items in the panel to update the selected items here. If the panel is the one selecting/deselecting, can't it just update the bIsSelected field in the item directly?

Reply:
Well yes, technically right now the only way selection happens is by clicking on an item, but imagine in the future if we had a tool which called `GEditor->SelectEntities`, or for example once we implement the ability for the user to click on an entity in the viewport.

This could definitely be improved but I think the simplicity of the code is worth more than the slight performance gain (for now).